packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1873,6 +1873,12 @@ export class Bucket extends BucketBase {
       throw new Error("Cannot use 'noncurrent' rules on a nonversioned bucket");
     }
 
+    rule.transitions?.forEach((transition) => {
+      if ((transition.transitionAfter && transition.transitionDate) || (!transition.transitionAfter && !transition.transitionDate)) {
+        throw new Error('Exactly one of transitionDate and transitionAfter must be specified');
+      }
+    });
+
```

Comment:
I take it you really dislike throwing separate Errors with dedicated messages for the cases when either both are specified or both are not - is there a particular reason for that?

Reply:
I split it into two cases with different messages.